GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs

```diff
@@ -269,5 +213,80 @@ internal override void ComputePlayerCombatReplayActors(AbstractPlayer p, ParsedE
             replay.AddOverheadIcons(fixations, p, ParserIcons.FixationPurpleOverhead);
             replay.AddTether(fixationEvents, Colors.Magenta, 0.5);
         }
+
+        internal override void ComputeEnvironmentCombatReplayDecorations(ParsedEvtcLog log)
+        {
+            base.ComputeEnvironmentCombatReplayDecorations(log);
+
+            // Vile Spit - Indicators
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.SiaxVileSpitIndicator, out IReadOnlyList<EffectEvent> vileSpitIndicators))
+            {
+                foreach (EffectEvent effect in vileSpitIndicators)
+                {
+                    // Indicator effect has variable duration
+                    (long start, long end) lifespan = (effect.Time, effect.Time + effect.Duration);
+                    EnvironmentDecorations.Add(new CircleDecoration(240, lifespan, Colors.LightOrange, 0.2, new PositionConnector(effect.Position)));
+                }
+            }
+
+            // Vile Spit - Poison
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.SiaxVileSpitPoison, out IReadOnlyList<EffectEvent> vileSpitPoisons))
+            {
+                foreach (EffectEvent effect in vileSpitPoisons)
+                {
+                    (long start, long end) lifespan = effect.ComputeDynamicLifespan(log, 15600);
+                    EnvironmentDecorations.Add(new CircleDecoration(240, lifespan, Colors.GreenishYellow, 0.2, new PositionConnector(effect.Position)));
+                }
+            }
+
+            // Nightmare Hallucinations Spawn Event
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.SiaxNightmareHallucinationsSpawnIndicator, out IReadOnlyList<EffectEvent> spawnEffects))
+            {
+                int duration = 3000;
+                foreach (EffectEvent effect in spawnEffects)
+                {
+                    var circle = new CircleDecoration(360, (effect.Time, effect.Time + duration), Colors.Orange, 0.2, new PositionConnector(effect.Position));
+                    EnvironmentDecorations.Add(circle);
+                    EnvironmentDecorations.Add(circle.Copy().UsingGrowingEnd(effect.Time + duration));
+                }
+            }
+
+            // Caustic Barrage
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.CausticBarrageIndicator, out IReadOnlyList<EffectEvent> barrageEffects))
```

Comment:
Barring a few numerical values, this is exactly the same code as in Ensolyss, can be refactored.